src/TopPage/FluentExtension.php

```diff
@@ -34,7 +36,7 @@ class FluentExtension extends DataExtension
      */
     protected function assignTopPage(Page $page): void
     {
-        parent::assignTopPage($page);
+        TopPageTrait::assignTopPage($page);
```

Comment:
Calling a static method on a trait will cause a deprecation warning in php8.1
https://www.php.net/manual/en/migration81.deprecated.php#migration81.deprecated.core.static-trait

You'll need to update this to call something like `ElementalArea::assignTopPage($page)`

Reply:
If we went with an extension this could be an instance method using `$this->owner->assignTopPage($page)`